main.go

```diff
@@ -448,6 +448,25 @@ func collectStructs(node ast.Node) map[token.Pos]*structType {
 		case *ast.ValueSpec:
 			structName = x.Names[0].Name
 			t = x.Type
+		case *ast.Field:
+			if len(x.Names) > 0 {
+				structName = x.Names[0].Name
+			}
+			t = x.Type
+		}
+
+		// if found expression is a "*something" or "[]something",
+		// dereference to check if "something" contains a struct expression
+	loop:
+		for true {
+			switch x := t.(type) {
+			case *ast.StarExpr:
+				t = x.X
+			case *ast.ArrayType:
+				t = x.Elt
+			default:
+				break loop
+			}
```

Comment:
I'm not a huge fan of goto's. Can you please change this so it doesn't use this pattern? You can create a small function that would return `t`

Reply:
(I completely missed your notification) updated